version: '3.5'

services:

  postgres:
    build:
      context: .
      dockerfile: Dockerfile-postgres
    container_name: postgres
    networks:
      - p8e
    volumes:
      - postgres:/var/lib/postgresql/data
    environment:
      - POSTGRES_USER=postgres
      - POSTGRES_PASSWORD=password1
    ports:
      - 5432:5432
    healthcheck:
      test: [CMD-SHELL, pg_isready -U postgres]
      interval: 5s
      timeout: 10s
      retries: 5
    command: postgres

  jaeger:
    image: jaegertracing/all-in-one:latest
    container_name: jaeger
    networks:
      - p8e
    ports:
      - 6831:6831
      - 5775:5775/udp
      - 16686:16686

  object-store:
    image: ghcr.io/provenance-io/object-store:0.2.0
    container_name: object-store
    networks:
      - p8e
    depends_on:
      - postgres
    environment:
      - OS_URL=0.0.0.0
      - OS_PORT=8080
      - DB_CONNECTION_POOL_SIZE=3
      - DB_HOST=postgres
      - DB_USER=postgres
      - DB_PASS=password1
      - DB_PASSWORD=password1
      - DB_PORT=5432
      - DB_NAME=object-store
      - DB_SCHEMA=public
      - URI_HOST=localhost:8080
      - STORAGE_TYPE=file_system
      - STORAGE_BASE_PATH=/mnt/data
      - RUST_LOG=warn,object_store=debug
    ports:
      - 5000:8080
    volumes:
      - object_store:/mnt/data

  provenance:
    image: provenanceio/provenance:v1.5.0
    container_name: provenance
    command: bash -c "cp -rn /home/provenance_seed/* /home/provenance && /usr/bin/provenanced -t --home /home/provenance start"
    networks:
      - p8e
    ports:
      - 1317:1317
      - 9090:9090
      - 26657:26657
    volumes:
      - ./data/provenance:/home/provenance_seed:ro
      - provenance:/home/provenance

volumes:
  postgres:
  provenance:
  object_store:

networks:
  p8e:
    ipam:
      driver: default
      config:
        - subnet: 172.19.10.0/24
